src/playlist-loader.js

```diff
@@ -230,16 +230,24 @@ export default class PlaylistLoader extends EventTarget {
 
     // merge this playlist into the master
     const update = updateMaster(this.master, parser.manifest);
-    let refreshDelay = (parser.manifest.targetDuration || 10) * 1000;
 
     this.targetDuration = parser.manifest.targetDuration;
+
+    // if the playlist is unchanged since the last reload or last segment duration cannot
+    // be determined, try again after half the target duration
+    let refreshDelay = (this.targetDuration || 10) * 500;
+
     if (update) {
       this.master = update;
       this.media_ = this.master.playlists[parser.manifest.uri];
+
+      // the client MUST wait for at least the duration of the last segment in the
+      // Playlist before attempting to reload the Playlist file again
+      if (this.media_.segments.length) {
```

Comment:
If there are no media segments, should we be more or less aggressive than half the target duration? Is there any recommendation of what to do in that case from Apple, or what Safari does?

Reply:
It only has the recommendation of if the playlist is unchanged, to do half of target duration. It also has this statement
```
However the client MUST NOT attempt to reload the Playlist file more
   frequently than specified by this section, in order to limit the
   collective load on the server.
```
So I think it we should not be more aggressive and just stick with the half target duration when we encounter strange edge cases